mbuild/packing.py

```diff
@@ -57,12 +74,31 @@ def fill_box(compound, n_compounds, box, overlap=0.2, seed=12345):
                          "packmol.exe is on the path.")
         raise IOError(msg)
 
-    box = _validate_box(box)
+    arg_count = 3 - [n_compounds, box, density].count(None)
+    if arg_count != 2:
+        msg = "Exactly 2 of n_compounds, box, and density must be specified. {} were given.".format(arg_count)
+        raise ValueError(msg)
+
+    if box is not None:
+        box = _validate_box(box)
     if not isinstance(compound, (list, set)):
         compound = [compound]
-    if not isinstance(n_compounds, (list, set)):
+    if n_compounds is not None and not isinstance(n_compounds, (list, set)):
         n_compounds = [n_compounds]
 
+    if density is not None:
+        if box is None and n_compounds is not None:
+            total_mass = np.sum([n*np.sum([a.mass for a in c.to_parmed().atoms]) for c,n in zip(compound, n_compounds)])
+            L = (total_mass/density)**(1/3)*1.1841763 # Conversion from amu/(kg/m^3) to nm
+            box = _validate_box(Box(3*[L]))
+        if n_compounds is None and box is not None:
+            if len(compound) > 1:
+                msg = "Determing n_compounds from system density currently supports systems with only one compound type"
```

Comment:
Wrap to 80. Also would this be clearer if it said something along the lines of "Determining `n_compounds` from `density` + `box` currently only supported for systems with one compound type"?